src/services/dashboards/widgets/aws-cloud-front-cost/AWSCloudFrontCost.vue

```diff
@@ -39,122 +45,217 @@ import {
-import { GROUP_BY } from '@/services/dashboards/widgets/config';
-import type { HistoryDataModel } from '@/services/dashboards/widgets/type';
+import { WIDGET_SIZE } from '@/services/dashboards/widgets/config';
+import type { HistoryDataModel, Legend } from '@/services/dashboards/widgets/type';
 import { useWidgetFrameProps } from '@/services/dashboards/widgets/use-widget-frame-props';
 import { useWidgetLifecycle } from '@/services/dashboards/widgets/use-widget-lifecycle';
 // eslint-disable-next-line import/no-cycle
 import { useWidgetState } from '@/services/dashboards/widgets/use-widget-state';
 import { GROUP_BY_ITEM_MAP } from '@/services/dashboards/widgets/view-config';
+import { getLegends, getRefinedXYChartData } from '@/services/dashboards/widgets/widget-chart-helper';
+import { sortTableData } from '@/services/dashboards/widgets/widget-table-helper';
+
 
-// TODO: sample data
-const SAMPLE_RAW_DATA = {
-    more: true,
-    results: range(5).map((d) => ({
-        project_id: `project${d + 1}`,
-        'data-transfer.out': random(100, 1000),
-        'requests.http': random(1000, 2000),
-        'requests.https': random(100, 5000),
-    })),
+type Data = HistoryDataModel['results'];
+interface FullData {
+    results: Data;
+    more: boolean;
+}
+type UsageType = 'data-transfer.out' | 'requests.http' | 'requests.https';
+
+const USAGE_SOURCE_UNIT = 'GB';
+const SELECTOR_TYPE_FIELDS_KEY_MAP = {
+    cost: 'usd_cost',
+    usage: 'usage_quantity',
+};
+const USAGE_TYPE_LABEL_MAP: Record<UsageType, string> = {
+    'data-transfer.out': 'Transfer-out',
+    'requests.http': 'Requests (HTTP)',
+    'requests.https': 'Requests (HTTPS)',
 };
-const CATEGORY_FIELD_NAME = GROUP_BY.PROJECT;
 
 const props = defineProps<WidgetProps>();
 const chartContext = ref<HTMLElement|null>(null);
-const {
-    createXYVerticalChart, createXYColumnSeries, createLegend,
-    createTooltip, setXYSharedTooltipText, disposeRoot, refreshRoot,
-    // setChartColors,
-} = useAmcharts5(chartContext);
+const chartHelper = useAmcharts5(chartContext);
 
 const state = reactive({
-    ...toRefs(useWidgetState<HistoryDataModel['results']>(props)),
-    labels: computed(() => ['data-transfer.out', 'requests.http', 'requests.https']),
-    tableFields: computed(() => [
-        GROUP_BY_ITEM_MAP[GROUP_BY.PROJECT],
-        { name: 'data-transfer.out', label: 'Transfer-Out' },
-        { name: 'requests.http', label: 'Requests (HTTP)' },
-        { name: 'requests.https', label: 'Requests (HTTPS)' },
-    ]),
+    ...toRefs(useWidgetState<FullData>(props)),
+    fieldsKey: computed<string>(() => SELECTOR_TYPE_FIELDS_KEY_MAP[state.selectedSelectorType]),
+    legends: [] as Legend[],
+    chartData: computed(() => {
+        const valueKey = `${state.fieldsKey}_sum`;
+        const _chartData = getRefinedXYChartData(state.data?.results, state.groupBy, 'usage_type', valueKey, true);
+        return reverse(_chartData);
```

Comment:
nitpick: It can be replaced with Array.prototype.reverse()

```suggestion
        return_chartData.reverse();
```